nixos/modules/services/hardware/libinput.nix

```diff
@@ -18,26 +18,96 @@ let cfg = config.services.libinput;
       };
 
       accelProfile = mkOption {
-        type = types.enum [ "flat" "adaptive" ];
+        type = types.enum [ "flat" "adaptive" "custom" ];
         default = "adaptive";
         example = "flat";
         description = ''
             Sets the pointer acceleration profile to the given profile.
-            Permitted values are `adaptive`, `flat`.
+            Permitted values are `adaptive`, `flat`, `custom`.
             Not all devices support this option or all profiles.
             If a profile is unsupported, the default profile for this is used.
             `flat`: Pointer motion is accelerated by a constant
             (device-specific) factor, depending on the current speed.
             `adaptive`: Pointer acceleration depends on the input speed.
             This is the default profile for most devices.
+            `custom`: Allows the user to define a custom acceleration function.
+            To define custom functions use the accelPoints<Fallback/Motion/Scroll>
+            and accelStep<Fallback/Motion/Scroll> options.
           '';
       };
 
       accelSpeed = mkOption {
         type = types.nullOr types.str;
         default = null;
         example = "-0.5";
-        description = "Cursor acceleration (how fast speed increases from minSpeed to maxSpeed).";
+        description = ''
+            Cursor acceleration (how fast speed increases from minSpeed to maxSpeed).
+            This only applies to the flat or adaptive profile.
+          '';
+      };
+
+      accelPointsFallback = mkOption {
+        type = types.nullOr types.str;
```

Comment:
```suggestion
        type = types.nullOr (types.listOf types.number)
```
Ditto for below

Reply:
I can see the reason for changing the types, and had considered it. I prioritized consistency with the pre-existing options, as there are already very similar options in the same module that are typed as `str`. Would it be better to ignore them?